internal/impl/generate/devcontainer_util.go

```diff
@@ -23,6 +23,15 @@ import (
 //go:embed tmpl/*
 var tmplFS embed.FS
 
+type Generate struct {
+	Ctx            context.Context
+	Path           string
+	RootUser       bool
+	IsDevcontainer bool
+	Pkgs           []string
+	LocalFlakeDirs []string
+}
```

Comment:
A few issues:

* Not sure about the name (`generate.Generate` is repetitive and doesn't convey what it does).
* As currently used it does not need to be exported.
* It should not contain context. (In general context should never be saved in a struct).

Proposal:

* Name it `Options` instead.
* Remove context.
* Remove Open

Code would look like:

```golang
g := &generate.Options{...}
g.CreateDockerfile(ctx)
```